### ISSUE TITLE
Guillermo's Open Project Ideas

### ISSUE THREAD

**Issue by lealgamb** [vocatwk/CodeU-Nemo#32]
Here are some of the ideas that interest me:

- Add the ability to @mention a user.

- Implement a notification system.

- Create a search index from the message database.

- Write a bot that generates new conversations and messages, and make it one of your users.

Let me know what you guys think. Also, ideally I'd be working on adding features to the activity feed and cleaning up the code along the way.

**Comment by vocatwk** [comment]
All sounds great. Third one would be particularly helpful in all the other features we could implement!

Not sure if I understand the fourth one though.

**Comment by lealgamb** [comment]
I'll make sure that's the first feature I work on. I took the last one from the suggested ideas. I also don't fully understand it but I thought it would be interesting to work on, or at least attempt.

**Comment by vocatwk** [comment]
yeah I just looked up what bots actually do and then edited the message lol

**Comment by yebrahim** [comment]
The bot idea is very interesting. It's one of those features that can be as complicated and involved as we want it to be.
Do you guys have any thoughts on what it would look like?

**Comment by lealgamb** [comment]
Honestly I'm not too sure. Are you asking what it would do or something else?

**Comment by yebrahim** [comment]
Yea what do you think it should do? Is is like a personal assistant? What kind of questions can you ask it?

**Comment by lealgamb** [comment]
Hmm. Perhaps making it a personal assistant would be useful. Instead of having a FAQ the bot can take its place. I'm thinking that if we implement styled text then users can ask it for the necessary symbols.

**Comment by lealgamb** [comment]
I created a Team Drive and shared the doc on there.

**Comment by yebrahim** [comment]
Here's an idea to kick off a simple bot feature: If users send a message to any conversation, where the message has some special format (e.g. starts with "/bot", "/assistant", "/help".. etc), then the message is sent to a new `Bot` class, which handles the message content. For now, we can have it respond by saying "hi, this is your assistant" or something, to every message.

The goal is to try to remove much of the complexity that comes with the idea of implementing a chatbot to make it less daunting, so we can get started quickly, while at the same time leaving a lot of gaps for improvement.

Thoughts?

**Comment by vocatwk** [comment]
I like it. we could make it so that it so that it goes to the Bot class when they @ mention the Bot.

**Comment by yebrahim** [comment]
Yes. In this case, the bot is available to all users in all conversations, which is a simple way to start the feature. Later on when we have group messaging, we can have users add bots to conversations before they start @ mentioning them.

**Comment by lealgamb** [comment]
Sounds good. Would we parse the message right after it's received and stored in the MessageStore?

**Comment by yebrahim** [comment]
That seems like a good place to start, yes.

**Comment by yebrahim** [comment]
Should we start a new issue to discuss the details of the bot as we go? I'd start with listing the feature requirements, what you think the bot should and shouldn't do, without diving into its design or architecture.

**Comment by lealgamb** [comment]
@yebrahim will do 👍
